test/CodeGen/AVR/smul-with-overflow.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc < %s -march=avr | FileCheck %s
+
+define i1 @signed_multiplication_did_overflow(i8, i8) unnamed_addr {
+; CHECK-LABEL: signed_multiplication_did_overflow:
```

Comment:
Something like this would be good (with patterns instead of registers):

``` llvm
define i1 @ signed_multiplication_did_overflow(i8, i8) unnamed_addr {
; CHECK-LABEL: signed_multiplication_did_overflow:
entry-block:
  %2 = tail call { i8, i1 } @llvm.umul.with.overflow.i8(i8 %0, i8 %1)
  %3 = extractvalue { i8, i1 } %2, 1
  ret i1 %3
; CHECK-NEXT: muls r24, r22
; CHECK-NEXT: mov r25, r1
; ...
}

declare { i8, i1 } @llvm.umul.with.overflow.i8(i8, i8)
```

Of course, don't worry about any calling convention or prologue/epilogue junk.

Same with the other test btw